src/test/java/org/mockitousage/matchers/CapturingArgumentsTest.java

```diff
@@ -314,4 +330,53 @@ public void captures_correctly_when_captor_used_on_pure_vararg_method() throws E
         verify(mock).varargs(eq(42), argumentCaptor.capture());
         Assertions.assertThat(argumentCaptor.getValue()).contains("capturedValue");
     }
+
+    @Test
+    public void should_capture_only_list_instances() throws Exception {
+        //given
+        ArgumentCaptor<List> captor = ArgumentCaptor.forClass(List.class);
+
+        //when
+        mock.forCollection(new ArrayList<String>());
+        mock.forCollection(new HashSet<String>());
+        mock.forCollection(new LinkedList<String>());
+
+        //then
+        verify(mock, times(2)).forCollection(captor.captureIf(new InstanceOf(List.class)));
+        assertThat(2, is(equalTo(captor.getAllValues().size())));
+    }
+
+    @Test
+    public void should_capture_when_value_is_equal_to_a_certain_value() throws Exception {
+        //given
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+
+        //when
+        mock.simpleMethod(FOO, 1);
+        mock.simpleMethod(BAR, 2);
+
+        //then
+        verify(mock).simpleMethod(captor.captureIf(new Equals(FOO)), anyInt());
+        assertThat(1, is(equalTo(captor.getAllValues().size())));
+        assertThat(FOO, is(equalTo(captor.getValue())));
+    }
+
+    @Test
+    public void should_capture_when_value_is_less_then() {
+        //given
+        ArgumentCaptor<Integer> captor = ArgumentCaptor.forClass(Integer.class);
+
+        //when
+        mock.simpleMethod(FOO, 1);
+        mock.simpleMethod(BAR, 2);
+        mock.simpleMethod(FOO, 3);
+        mock.simpleMethod(BAR, 4);
+        
+        mock.simpleMethod(FOO, 5);
+
+        //then
+        verify(mock, times(2)).simpleMethod(anyString(), captor.captureIf(new LessThan<Integer>(3)));
```

Comment:
@terebesirobert You are right we can't call `captureIf(isA(XY.class))`. I totally forgot that `AdditionalMatchers.*` and `Matchers.*` return default values and no `ArgumentMatchers`. Adding specializations like  `captureIfIsA(..)` or `caputureIfGreaterThat(..)` would add too much complexity and would exclude custom ArgumentMatcher implementations.

What do you think of a class like _ArgumentMatcher_**s** that contains all matchers known from AdditionalMatchers and Matcher but return the ArgumentMatcher instead default values. Eg.:

```
class ArgumentMatchers{
   <T> ArgumentMatcher<T> isA(Class<? extends T> type){
        return new InstanceOf(type);
   }
  [..]
}
```